RandLAPACK/drivers/rl_cqrrp.hh

```diff
@@ -372,8 +358,8 @@ int CQRRP_blocked<T, RNG>::call(
         // Checking for the zero matrix post-pivoting is the best idea, 
         // as we would only need to check one column (pivoting moves the column with the largest norm upfront)
         block_zero = true;
-        for (size_t i = 0; i < rows; ++i) {
-            if (A_work[i] != 0.0) {
+        for (i = 0; i < rows; ++i) {
+            if (A_work[i] > std::numeric_limits<T>::epsilon()) {
```

Comment:
I think you need an abs around A_work[i].

Reply:
Fixed.